main/src/hadd.cxx

```diff
@@ -348,62 +381,127 @@ int main( int argc, char **argv )
          std::cout << "hadd compression setting for all ouput: " << newcomp << '\n';
    }
    if (append) {
-      if (!merger.OutputFile(targetname,"UPDATE",newcomp)) {
+      if (!fileMerger.OutputFile(targetname,"UPDATE",newcomp)) {
          std::cerr << "hadd error opening target file for update :" << argv[ffirst-1] << "." << std::endl;
          exit(2);
       }
-   } else if (!merger.OutputFile(targetname,force,newcomp) ) {
+   } else if (!fileMerger.OutputFile(targetname,force,newcomp) ) {
       std::cerr << "hadd error opening target file (does " << argv[ffirst-1] << " exist?)." << std::endl;
       if (!force) std::cerr << "Pass \"-f\" argument to force re-creation of output file." << std::endl;
       exit(1);
    }
 
+   auto filesToProcess = argc - ffirst;
+   auto step = (filesToProcess+nProcesses-1)/nProcesses;
+   if(step < 3){
+      //At least 3 files per process
+      step = 3;
+      nProcesses = (filesToProcess+step-1)/step;
+      std::cout<<"Each process should handle at least 3 files for efficiency.";
+      std::cout<<" Setting the number of processes to: "<<nProcesses<<std::endl;
+   }
+   std::vector<std::string> partialFiles;
+   
+   if(multiproc){
+      for(auto i = 0; (i*step)<filesToProcess; i++) {
+         std::stringstream buffer;
+         buffer <<"partial"<<i<<".root";
```

Comment:
If I read correctly the intermediary files name are "partial1.root", "partial2.root".   If this is the case, we may want to enhance the name to allow for the possibility of running two hadd in the same directory at the same time (i.e. at the moment this is not possible).